pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTNumericLiteral.java

```diff
@@ -125,17 +125,26 @@ public int getBase() {
 
     static int getBase(Chars image) {
         if (image.length() > 1 && image.charAt(0) == '0') {
+            if (image.indexOf('.') != -1) {
+                return 10;
+            }
```

Comment:
this is actually not true…

The value `0x1.0p+2f` is an hexadecimal float literal in Java. It's still base 16.

I'm seeing a lot of false positives in the `UseUnderscoresInNumericLiterals` rule because of this changeset.

Reply:
Moved this check to after the binary and hex checks.